qa/L0_DALI_GPU_ensemble/model_repository/dali_1/pipeline.py

```diff
@@ -0,0 +1,43 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2021 NVIDIA CORPORATION
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import nvidia.dali as dali
+import nvidia.dali.fn as fn
+import multiprocessing as mp
+import argparse
+
+@dali.pipeline_def(batch_size=1, num_threads=min(mp.cpu_count(), 4), device_id=0)
```

Comment:
Why? This batch size isn't relevant for the DALI backend, right?

Reply:
It sort of is - indeed, it would be best to check that it works with _variable_ batch size, since by far Triron is the only client of DALI that uses variable batch size though C API.

Reply:
Does it really matter? This batch size is discarded anyway.

Reply:
AThis `batch_size`  parameter is discarded when we deserialize pipeline and it's assigned the `max_batch_size` parameter from the config. In the tests, the batch size is variable.

Reply:
OK.